ddl: add description of ddl v2 (#16073)
pingcap/docs: ddl-v2.md

---
title: Use TiDB DDL V2 to Accelerate Table Creation
summary: Learn the concept, principles, and implementation details of TiDB DDL V2 for acceleration table creation.
---

# Use TiDB DDL V2 to Accelerate Table Creation

Starting from v7.6.0, the new version V2 of TiDB DDL supports creating tables quickly, which improves the efficiency of bulk table creation.

TiDB uses the online asynchronous schema change algorithm to change the metadata. All DDL jobs are submitted to the `mysql.tidb_ddl_job` table, and the owner node pulls the DDL job to execute. After executing each phase of the online DDL algorithm, the DDL job is marked as completed and moved to the `mysql.tidb_ddl_history` table. Therefore, DDL statements can only be executed on the owner node and cannot be linearly extended.

However, for some DDL statements, it is not necessary to strictly follow the online DDL algorithm. For example, the `CREATE TABLE` statement only has two states for the job: `none` and `public`. Therefore, TiDB can simplify the execution process of DDL, and executes the `CREATE TABLE` statement on a non-owner node to accelerate table creation.

> **Warning:**
>

> > This variable is currently an experimental feature and it is not recommended to use in a production environment. This feature might change or be removed without prior notice. If you find a bug, please give feedback by raising an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

Reviewer: ```suggestion
> This feature is currently an experimental feature and it is not recommended to use in a production environment. This feature might change or be removed without prior notice. If you find a bug, please give feedback by raising an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
```

## Compatibility with TiDB tools

- [TiCDC](/ticdc/ticdc-overview.md) does not support replicating the tables that are created by TiDB DDL V2.

## Limitation

You can now use TiDB DDL V2 only in the [`CREATE TABLE`](/sql-statements/sql-statement-create-table.md) statement, and this statement must not include any foreign key constraints.

> ## How to use TiDB DDL V2

Reviewer: ```suggestion
## Use TiDB DDL V2
```

You can enable or disable TiDB DDL V2 by specifying the value of the system variable [`tidb_ddl_version`](/system-variables.md#tidb_ddl_version-new-in-v760) .

To enable TiDB DDL V2, set the value of this variable to `2`:

```sql
SET GLOBAL tidb_ddl_version = 2;
```

To disable TiDB DDL V2, set the value of this variable to `1`:

```sql
SET GLOBAL tidb_ddl_version = 1;
```

> ## Implementation principle of TiDB DDL V2

Reviewer: ```suggestion
## Implementation principle
```

---
title: Use TiDB DDL V2 to Accelerate Table Creation
summary: Introduce the concept, principles, and implementation details of TiDB DDL V2 on acceleration table creation.
---

# Use TiDB DDL V2 to Accelerates Table Creation

Starting from v7.6.0, the new version V2 of TiDB DDL supports creating tables quickly, which improves the efficiency of bulk table creation.

> **Warning:**
>
> This variable is currently an experimental feature and it is not recommended to use in a production environment. This feature might change or be removed without prior notice. If you find a bug, please give feedback by raising an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

## Compatibility with TiDB tools

> - [TiCDC](/ticdc/ticdc-overview.md) does not support using TiDB DDL V2 to create tables.

Reviewer 2: ```suggestion
- [TiCDC](/ticdc/ticdc-overview.md) does not support replicating the tables which are created by TiDB DDL V2.
```
